python/tk_desktop/rpc.py

```diff
@@ -191,16 +293,31 @@ def run(self):
                 # just keep serving new connections
                 pass
             finally:
-                logger.debug("server closing")
-                connection.close()
+                # It's possible we failed accepting, so the variable may not be defined.
+                if "connection" in locals():
```

Comment:
Why is it being done like this, rather than say defining `connection = None` just before the start of the `try` at line `236`?

Reply:
I think the technical term for doing things this way is called a "brainfart".